dali/python/nvidia/dali/pool.py

```diff
@@ -0,0 +1,463 @@
+# Copyright (c) 2020, NVIDIA CORPORATION. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import socket
+import threading
+import multiprocessing
+from collections import OrderedDict
+from nvidia.dali import backend as _b
+from nvidia.dali.worker import worker
+from nvidia.dali.messages import ScheduledTasks
+from nvidia.dali.shared_batch import SharedBatchMeta
+from nvidia.dali.shared_batch import deserialize_sample_meta, deserialize_batch, import_numpy
+from nvidia.dali import shared_mem
+
+
+class SharedBatchesConsumer:
+    """Counterpart of worker.py:SharedBatchesDispatcher. Can receive and deserialize batch
+    from the worker, keeps track of already received memory chunks and opens new chunks
+    or resizes exiting ones if necessary."""
+    class MemChunk:
+        def __init__(self, shm_chunk: shared_mem.SharedMem, capacity: int):
+            self.shm_chunk = shm_chunk
+            self.capacity = capacity
+
+    def __init__(self):
+        import_numpy()
+        self.batch_pool = {}
+
+    def get_mem_chunk(self, sock: socket.socket, batch: SharedBatchMeta) -> MemChunk:
+        """Get the fd for shared memory through sock and mmap the memory based on metadata
+        in `batch`. Adjust if it was previously mapped.
+
+        Parameters
+        ----------
+        `sock` : socket.socket
+            Socket used to transfer file descriptors.
+        `batch` : SharedBatchMeta
+            Serialized batch metadata
+
+        Returns
+        -------
+        MemChunk
+            Wrapped SharedMem object with current capacity containing the obtained data.
+        """
+        chunk = self.batch_pool.get(batch.mem_chunk_id)
+        if chunk is not None:
+            if chunk.capacity != batch.capacity:
+                chunk.shm_chunk.resize(batch.capacity, trunc=False)
+                chunk.capacity = batch.capacity
+            return chunk
+        fd, shm_chunk = -1, None
+        try:
+            [fd] = multiprocessing.reduction.recvfds(sock, 1)
```

Comment:
This is an OS-specific function. Use a portable one instead:
```suggestion
            handle = multiprocessing.reduction.recv_handle()
```

Reply:
Done. With for windows todo at the bottom. :man_shrugging: